### PR TITLE
Add new module renovate

### LINKED ISSUE #9060: Module: renovate
It might be nice to have a module to configure [renovate](https://github.com/renovatebot/renovate) for a jhipster-lite generated project, with e.g. docker-compose dependencies update for content in `src/main/docker` (as done in #9059) and good default settings.

### PR DESCRIPTION
**Pull request by swarajsaaj**
Add renovate for generated projects for automatic dependency upgdates

Fix #9060

### CODE REVIEW

File: src/main/resources/generator/ci/renovate/renovate.json.mustache [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  "extends": [
+    "config:recommended",
+    ":disableDependencyDashboard",
+    ":automergeMinor",
+    ":automergePr",
+    ":automergeRequireAllStatusChecks",
+    ":label(area: dependencies)"
+  ],
+  "prHourlyLimit": 20,
+  "enabledManagers": [
```
Comment (murdos) [inline src/main/resources/generator/ci/renovate/renovate.json.mustache:12]: I think we should rather avoid declaring explicitly managers, so they will be auto discovered.

File: src/test/java/tech/jhipster/lite/generator/ci/renovate/RenovateModuleFactoryTest.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import tech.jhipster.lite.module.domain.JHipsterModule;
+import tech.jhipster.lite.module.domain.JHipsterModulesFixture;
+import tech.jhipster.lite.module.domain.properties.JHipsterModuleProperties;
+
+@UnitTest
+@ExtendWith(MockitoExtension.class)
+class RenovateModuleFactoryTest {
+
+  @InjectMocks
+  private RenovateModuleFactory factory;
```
Comment (murdos) [inline src/test/java/tech/jhipster/lite/generator/ci/renovate/RenovateModuleFactoryTest.java:23]: There's no dependency here, so no need for Mockito, we can just create a new instance using the constructor: 
```suggestion
  private RenovateModuleFactory factory = new RenovateModuleFactory();
```

**Review by murdos — COMMENTED** [review]
Could you also declare this module in the fullapp that is generated on CI? https://github.com/jhipster/jhipster-lite/blob/main/tests-ci/generate.sh#L173
That way the module generation will also be tested in real conditions.

File: src/test/java/tech/jhipster/lite/generator/ci/renovate/RenovateModuleFactoryTest.java (hunk 2 above)
Comment (swarajsaaj) [reply src/test/java/tech/jhipster/lite/generator/ci/renovate/RenovateModuleFactoryTest.java]: Nice catch. Thanks.

**Comment by murdos** [comment]
Thanks @swarajsaaj :)

**Pull request merged**